### DISCUSSION TITLE
How to access graphql?

### DISCUSSION THREAD

**Discussion by neugartf** [supabase/supabase#6645] [Questions] [answered]
Hey!

I'm trying to access my graphql via `https://vwrhnfjzsrwlhdwjwfgx.supabase.co/graphql/v1`, which returns me:
```
{
  "message": "Could not find the public.graphql() function in the schema cache",
  "hint": "If a new function was created in the database with this name and parameters, try reloading the schema cache."
}
```

The error sounds like that the path is wrongfully mapping to the `public` scheme and not the `public_graphql` where that function exists.

Any ideas? 🤔

**Comment by olirice** [comment, marked as answer]
hey @neugartf
cross posting here from the support issue in case others have the same issue.

could you please try loading 

```html
<html>
  <head>
    <title>Supabase GraphQL</title>
    <link href="https://unpkg.com/graphiql/graphiql.css" rel="stylesheet" />
  </head>
  <body style="margin: 0;">
    <div id="graphiql" style="height: 100vh;"></div>

    <script
      crossorigin
      src="https://unpkg.com/react/umd/react.production.min.js"
    ></script>
    <script
      crossorigin
      src="https://unpkg.com/react-dom/umd/react-dom.production.min.js"
    ></script>
    <script
      crossorigin
      src="https://unpkg.com/graphiql/graphiql.js"
    ></script>

    <script>
      const fetcher = GraphiQL.createFetcher({
         url: 'https://vwrhnfjzsrwlhdwjwfgx.supabase.co/graphql/v1',
         headers: {
             "apiKey": "your api key"
         }
      });
      ReactDOM.render(
        React.createElement(GraphiQL, { fetcher: fetcher }),
        document.getElementById('graphiql'),
      );
    </script>
  </body>
</html>
```

in your browser, subbing in your `apiKey` and let me know if you still see that error?

**Reply by neugartf** [reply]
Thanks both for your help!
Eventually the problem was that the `Continents` example in the admin interface contains enums with empty spaces, which is not supported. https://github.com/supabase/pg_graphql/issues/172. 
To get this object out of the db I had to manually connect to the db, it wasn't possible within the web interface.
Thanks for the quick help!

**Comment by zernonia** [comment]
Have a look at [Graphql is now available in Supabase](https://supabase.com/blog/2022/03/29/graphql-now-available) post. In the **Getting Started** section, it shows how to enable Graphql endpoint.

**Reply by olirice** [reply]
thank you for pointing out the schema prefix, I had missed that.

Under `Settings > API > Schema` do you have an entry for `graphql_public` like in the screenshot below?

<img width="856" alt="Screen Shot 2022-05-03 at 4 20 10 AM" src="https://user-images.githubusercontent.com/12958657/166430492-daa52544-acf8-4502-bed1-4576fda1d5da.png">

**Reply by jtkeyva** [reply]
I don't see it in my schema. How do I enable graphQl on an older supabase account? Thx

**Reply by olirice** [reply]
hi @jtkeyva 
While GraphQL is in beta it is only available on new projects AO March 2022. One option if you'd like to use graphql on an existing project is to follow the [migrating between projects](https://supabase.com/docs/guides/database#migrating-between-projects) guide to move your project to a new instance that has GraphQL available

**Reply by jtkeyva** [reply]
@olirice got it thanks!
